python/ray/object_store_client.py

```diff
@@ -0,0 +1,485 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import time
+
+import pyarrow
+import pyarrow.plasma as plasma
+# TODO: Cleanup imports after implement `serialization` module.
+import ray.cloudpickle as pickle
+from ray.exceptions import (RayTaskError, RayGetArgumentError, RayGetError)
+import ray.plasma
+from ray.serialization import CloudPickleError, RayNotDictionarySerializable
+from ray.services import logger
+import ray.signature
+import ray.ray_constants as ray_constants
+from ray.utils import thread_safe_client
+import ray.worker
+
+
+class ObjectStoreClient(object):
+    def __init__(self, worker, memcopy_threads=12):
+        """Data client for getting & putting objects.
+
+        Args:
+            worker: The worker instance.
+            memcopy_threads (int):
+                The number of threads Plasma should use when putting an object
+                in the object store.
+        """
+        self.worker = worker
+        self.memcopy_threads = memcopy_threads
+        # A dictionary that maps from driver id to SerializationContext
+        # TODO: clean up the SerializationContext once the job finished.
+        self.serialization_context_map = {}
+
+    def connect(self, store_socket_name, manager_socket_name, release_delay):
+        """Create a connection to plasma client.
+
+        Args:
+            store_socket_name (str):
+                Name of the socket the plasma store is listening at.
+            manager_socket_name (str):
+                Name of the socket the plasma manager is listening at.
+            release_delay (int):
+                The maximum number of objects that the client will keep and
+                delay releasing (for caching reasons).
+
+        """
+
+        if not self.use_raylet:
+            self.plasma_client = thread_safe_client(
+                plasma.connect(store_socket_name, manager_socket_name,
+                               release_delay))
+        else:
+            self.plasma_client = thread_safe_client(
+                plasma.connect(store_socket_name, "", release_delay))
+
+    def disconnect(self):
+        if hasattr(self, 'plasma_client'):
```

Comment:
we can initialize `plasma_client` as `None` in `__init__`, so it's more clear what attributes this class can have.